lms/djangoapps/verify_student/models.py

```diff
@@ -1037,29 +1036,27 @@ def set_deadline(cls, course_key, deadline, is_explicit=False):
                 record.save()
 
     @classmethod
-    def deadlines_for_courses(cls, course_keys):
+    def deadlines_for_enrollments(cls, enrollments_qs):
         """
-        Retrieve verification deadlines for particular courses.
+        Retrieve verification deadlines for a user's enrolled courses.
 
         Arguments:
-            course_keys (list): List of `CourseKey`s.
+            enrollments_qs: CourseEnrollment queryset. For performance reasons
```

Comment:
Would it be better to just get the user as an argument here? It would make the api less flexible, but also a bit more foolproof.

Reply:
I originally had it that way, but `student` already depends on `verify_student`, and I didn't want to introduce a circular dependency in the other direction by importing `CourseEnrollment` from `verify_student`. I could avoid the most obvious errors by doing a method-local import, but it still seemed icky.

Part of reason I left the method taking an enrollment queryset instead of passing in `.values('course_id')` is to make this method start exploding more obviously if it ever gets a list of enrollments.

Reply:
FWIW, I do think that the Right Thing would be a larger refactoring that makes this not a circular dependency, but that's way out of scope for what I want to do today. :)

Reply:
Oh wait, do users have `.enrollments`? Sigh, I never remember all the things we have hanging off models.

Reply:
Okay, looking at it again, it does look like I'd have to import `CourseEnrollment`, so yeah, circular dependency.

Reply:
Because `user` is a `ForeignKey` on `CourseEnrollment`, you probably could do `user.course_enrollment_set` to get all of the enrollments for a user.

Reply:
Or maybe `courseenrollment_set`, based on https://docs.djangoproject.com/en/1.11/topics/db/queries/#following-relationships-backward

Reply:
I dunno. I feel like `user.courseenrollment_set.filter(is_active=True)` is a little more brittle and a little more magical than passing in the enrollments queryset via `CourseEnrollment.enrollments_for_user()`. Maybe this is another instance of our "make the interface a little simpler" vs. "make the implementation a little simpler" philosophical differences. That being said, I only have a mild preference for passing in the queryset. Do you feel strongly about this?

Reply:
Nope.